fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/domain/documents/LiquidationLetter.java

```diff
@@ -0,0 +1,113 @@
+    private static final int MAX_SIZE_CITY = 50;
+    private static final int MAX_SIZE_REGION = 50;
+    private static final int MAX_SIZE_POSTAL_CODE = 20;
+
+    public static void create(final Event event) {
+        FinancialDocument.createFinancialDocument(event,
+                reportEntry -> toJson(event, reportEntry),
+                "carta-de-liquidacao",
+                DOCUMENT_TYPE,
+                json -> json.get("notificationRef").getAsString());
+    }
+
+    private static JsonObject toJson(final Event event, final ReportEntry reportEntry) {
+        final JsonObject json = new JsonObject();
+
+        addPersonalInformation(json, event, event.getPerson());
+
+        final LocalDate notificationDate = new LocalDate();
+        final LocalDate dueDate = notificationDate.plusDays(30);
+        json.addProperty("notificationDate", notificationDate.toString(DATE_FORMAT));
+        json.addProperty("notificationRef", notificationRefernece(event, notificationDate));
+        json.addProperty("debtDescription", event.getDescription().toString());
+
+        json.addProperty("limitDate",  dueDate.toString(DATE_FORMAT));
+
+        final Money debtTotal = reportEntry.amount.add(reportEntry.interest);
+        json.addProperty("debtValue", reportEntry.amount.toPlainString());
+        json.addProperty("debtInterest", reportEntry.interest.toPlainString());
+        json.addProperty("debtTotal", debtTotal.toPlainString());
+        json.addProperty("debtInitialDate", reportEntry.dueDate.toString(DATE_FORMAT));
+
+        json.addProperty("presidentGender", "male");
+
+        final EventPaymentCodeEntry eventPaymentCodeEntry = EventPaymentCodeEntry.create(event, debtTotal);
+        eventPaymentCodeEntry.setDueDate(dueDate.plusDays(7));
+        final PaymentCode paymentCode = eventPaymentCodeEntry.getPaymentCode();
+        json.addProperty("mbEntity", paymentCode.getEntityCode());
+        json.addProperty("mbRef", paymentCode.getFormattedCode());
+
+
+        return json;
+    }
+
+    private static String notificationRefernece(final Event event, final LocalDate notificationDate) {
+        return Integer.toString(notificationDate.getYear()) + "DA/PP/" + event.getExternalId();
+    }
+
+    static void addPersonalInformation(final JsonObject json, final Event event, final Person person) {
+        final String clientTIN = ClientMap.uVATNumberFor(person);
+        final String[] address = toAddress(clientTIN, person);
+
+        json.addProperty("gender", person.isMale() ? "male" : "female");
+        json.addProperty("name", person.getName());
+        json.addProperty("addressLine1", address[0]);
+        json.addProperty("addressLine2", address[1]);
+        json.addProperty("addressLine3", address[2]);
+        json.addProperty("tin", clientTIN);
+        json.addProperty("username", person.getUsername());
+    }
+
+    private static String[] toAddress(final String clientTIN, final Person person) {
```

Comment:
the way this address is being built is identical to the one sent to SAP however this MAX_SIZE's are specific for that platform they may not apply for this purpose

Reply:
The intention is to produce an address as close to the one that appears on the official financial document.
In this case it is a letter that will be sent via e-mail, so the available space is also limited.